app/src/main/java/com/kenchen/capstonenewsapp/networking/RemoteApi.kt

```diff
@@ -1,42 +1,27 @@
 package com.kenchen.capstonenewsapp.networking
 
 import com.kenchen.capstonenewsapp.model.Article
-import com.kenchen.capstonenewsapp.model.response.GetNewsResponse
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
 
 const val BASE_URL = "https://newsapi.org/v2/"
 const val API_KEY = "7ac9410c8eac436d90d7b8f121b5db36"
 
+/**
+ * Logic for Api calls
+ * */
+
 class RemoteApi(private val remoteApiService: RemoteApiService) {
 
     /**
-     * get top headlines news by country code
+     * get top headlines news by country code with Coroutine (suspend)
      * */
-    fun getTopHeadlinesByCountry(
-        countryCode: String,
-        OnNewsReceived: (RemoteResult<List<Article>>) -> Unit,
-    ) {
-        remoteApiService.getTopHeadlinesByCountry(countryCode, API_KEY)
-            .enqueue(object : Callback<GetNewsResponse> {
-                override fun onResponse(
-                    call: Call<GetNewsResponse>,
-                    response: Response<GetNewsResponse>,
-                ) {
-                    val getNewsResponse = response.body()
-                    if (getNewsResponse == null) {
-                        OnNewsReceived(RemoteResult.Failure(NullPointerException("No response " +
-                                "body!")))
-                    } else {
-                        OnNewsReceived(RemoteResult.Success(getNewsResponse.articles))
-                    }
-                }
-
-                override fun onFailure(call: Call<GetNewsResponse>, error: Throwable) {
-                    OnNewsReceived(RemoteResult.Failure(error))
-                }
+    suspend fun getTopHeadlinesByCountry(countryCode: String): List<Article> =
+        try {
+            val data = remoteApiService.getTopHeadlinesByCountry(countryCode, API_KEY)
+            val shuffledNews = data.articles.shuffled()
+            shuffledNews
```

Comment:
This is awesome! Great job randomizing the list with `shuffled()`. Note you can also just call the method without setting it to a val.  Much like lambda blocks, the `try` block will automatically use the result from the last statement as its return value:

```suggestion
            data.articles.shuffled()
```